hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/Partitioner.java

```diff
@@ -18,10 +18,14 @@
 
 package org.apache.hudi.table.action.commit;
 
+import org.apache.hudi.table.WorkloadProfile;
+
 import java.io.Serializable;
 
 public interface Partitioner extends Serializable {
   int getNumPartitions();
 
   int getPartition(Object key);
+
+  WorkloadProfile getExecutionWorkloadProfile();
 }
```

Comment:
Why a `Partitioner` returns the profile ? Let's not put the interface here.

Reply:
wondering if we can augment existing WorkloadProfile only to add another instance of HashMap<String, WorkloadStat> partitionPathStatMap.

```
  /**
   * Input workload profile.
   */
  protected final HashMap<String, WorkloadStat> inputPartitionPathStatMap;
  /**
   * Execution workload profile.
   */
  protected final HashMap<String, WorkloadStat> outputPartitionPathStatMap;
```

Let the UpsertPartitioner update the execution profile (a.k.a outputPartitionPathStatMap ) in the passed in workload profile itself. 
And so CommitActionExecutors does not need to poll partitionor for execution profile since they already have an handle to WorkloadProfile.